helpers/z_schema.js

```diff
@@ -43,72 +43,61 @@ var constants = require('./constants');
 
 var liskFormats = {
 	id: function (str) {
-		if (str.length === 0) {
-			return true;
-		}
-
-		return /^[0-9]+$/g.test(str);
+		return /^$/.test(str) || /^[0-9]+$/g.test(str);
 	},
 
 	additionalData: function (str) {
 		return Buffer.from(str).length <= constants.additionalData.maxLength;
 	},
 
 	address: function (str) {
-		if (str.length === 0) {
-			return true;
-		}
-
-		return /^[0-9]+[L]$/ig.test(str);
+		return /^$/.test(str) || /^[0-9]+[L]$/ig.test(str);
 	},
 
 	username: function (str) {
-		if (str.length === 0) {
-			return true;
-		}
-
-		return /^[a-z0-9!@$&_.]+$/ig.test(str);
+		return /^$/.test(str) || /^[a-z0-9!@$&_.]+$/ig.test(str);
 	},
 
 	hex: function (str) {
 		return /^[a-f0-9]*$/i.test(str);
 	},
 
 	publicKey: function (str) {
-		if (str.length === 0) {
-			return true;
-		}
-
-		return /^[a-f0-9]{64}$/i.test(str);
+		return /^$/.test(str) || /^[a-f0-9]{64}$/i.test(str);
 	},
 
 	csv: function (str) {
-		try {
-			var a = str.split(',');
-			if (a.length > 0 && a.length <= 1000) {
-				return true;
-			} else {
-				return false;
-			}
-		} catch (e) {
+		if (!str) {
 			return false;
 		}
-	},
 
-	signature: function (str) {
-		if (str.length === 0) {
+		var a = str.split(',');
+
+		if (a.length > 0 && a.length <= 1000) {
 			return true;
+		} else {
+			return false;
 		}
+	},
 
-		return /^[a-f0-9]{128}$/i.test(str);
+	signature: function (str) {
+		return /^$/.test(str) || /^[a-f0-9]{128}$/i.test(str);
 	},
 
 	queryList: function (obj) {
+		if (obj == null || obj == undefined) {
+			return false;
+		}
+
 		obj.limit = 100;
 		return true;
 	},
 
 	delegatesList: function (obj) {
+		if (obj == null || obj == undefined) {
```

Comment:
This would throw an error if number passed (line 112). Following would fit better here in my opinion
```
if (typeof obj !== 'object') {
   return false;
}
```